punishments.js

```diff
@@ -356,20 +381,85 @@ Punishments.lockRange = function (range, isIp) {
 	}
 };
 
+Punishments.roomBan = function (room, user, noRecurse, userid) {
+	if (!userid) userid = user.userid;
+	let alts;
+	let keys = noRecurse;
+	if (!keys || !keys.add) {
+		keys = new Set();
+	}
+	if (!noRecurse) {
+		alts = [];
+		keys = new Set();
+		Users.users.forEach(otherUser => {
```

Comment:
I'd seen this way of doing it used elsewhere as well and honestly forgot a little bit.